osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs

```diff
@@ -231,24 +145,27 @@ protected override void UpdateStartTimeStateTransforms()
         {
             base.UpdateStartTimeStateTransforms();
 
-            using (BeginDelayedSequence(-ring_appear_offset))
-                targetRing.ScaleTo(target_ring_scale, 400, Easing.OutQuint);
+            (spinnerBody.Drawable as ISkinnableSwell)?.AnimateSwellStart(this, MainPiece);
         }
 
         protected override void UpdateHitStateTransforms(ArmedState state)
         {
-            const double transition_duration = 300;
-
             switch (state)
             {
                 case ArmedState.Idle:
-                    expandingRing.FadeTo(0);
+                    // Only for rewind support. Reallows user inputs if swell is rewound from being hit/missed to being idle.
+                    HandleUserInput = true;
                     break;
 
                 case ArmedState.Miss:
                 case ArmedState.Hit:
-                    this.FadeOut(transition_duration, Easing.Out);
-                    bodyContainer.ScaleTo(1.4f, transition_duration);
+                    const int clear_animation_duration = 1200;
```

Comment:
So as far as I understand it, this is hardcoded to match the legacy animation. We generally do not do this; the `ISkinnableSwellPart` proposal could address this if you did something like `parts.Max(p => p.LatestTransformEndTime)` to set the lifetime end instead.

Reply:
I went through all your other review items, but this one is really giving me a hard time. Your proposal should definitely work in theory, but it doesn't because `LatestTransformEndTime` is not working for some reason. As a debug step I tried making things as simple as possible and just listing the transforms in `LegacySwellBody.Drawable` here, and it came up with 0 transforms (this explains why `LatestTransformEndTime` is not working). Do you have any idea what could be causing these transforms to not show up here? Could it be because I'm not transforming the entire `LegacySwellBody`, but only parts of it?

Reply:
I would have to go and check to be sure but that sounds plausible.